### DISCUSSION TITLE
Is it possible to inject the request context/request metadata into validators/validation decorators?

### DISCUSSION THREAD

**Discussion by mattrynbrand** [MichalLytek/type-graphql#1254] [General]
I'm building a multitenant application, we build up the database connections on application startup. When a request comes in we bind the relevant database connection based on the hostname to the context object. 

The issue is when a validation decorator needs to access the database (e.g checking for an object's existence), I'm currently unsure how I can inform the decorator of which database connection to use.

**Comment by MichalLytek** [comment]
I think class-validator was intended for a static validation, like `@IsEmail` or `@Lenght`.
If you need to query the db, it means it's your business logic and should be written as a normal code.
You can try then to abstract that with middlewares or something.

**Reply by mattrynbrand** [reply]
The `class-validator` docs have examples using database operations inside validators (https://github.com/typestack/class-validator#custom-validation-decorators) but its less of an issue if you only are dealing with a single database connection. But if there's not an existing way to do this I'll see if I can come up with some custom way to do it.

Appreciate your help and time Michał 🙏
